pkg/lib/_internal/dnf5daemon.ts

```diff
@@ -0,0 +1,309 @@
+                {
+                    package_attrs: { t: 'as', v: package_attrs },
+                    scope: { t: 's', v: "all" },
+                    patterns: { t: 'as', v: pkgnames }
+                }
+            ]) as ListPackage[][];
+
+            for (const pkg of results) {
+                if (seen_names.has(pkg.name.v))
+                    continue;
+
+                if (pkg.is_installed.v) {
+                    installed_names.add(pkg.name.v);
+                } else {
+                    data.missing_ids.push(pkg.id.v);
+                    data.missing_names.push(pkg.name.v);
+                }
+
+                seen_names.add(pkg.name.v);
+            }
+
+            pkgnames.forEach((name: string) => {
+                if (!installed_names.has(name) && data.missing_names.indexOf(name) == -1)
+                    data.unavailable_names.push(name);
+            });
+        }
+
+        async function simulate(session: string) {
+            if (data.missing_ids.length === 0 || data.unavailable_names.length > 0) {
+                return null;
+            }
+
+            await call(session, "org.rpm.dnf.v0.rpm.Rpm", "install", [pkgnames, {}]);
+            const [transaction_items, result] = await call(session, "org.rpm.dnf.v0.Goal", "resolve", [{}]) as InstallResolveResult;
+            if (result !== 0)
+                throw new ResolveError(`Resolving install failed with result={result}`);
+
+            for (const transaction_item of transaction_items) {
+                const [sort, _action, reason, _leftover, pkg] = transaction_item;
+                const name = pkg.name.v;
+
+                if (sort !== "Package")
+                    continue;
+
+                data.download_size += pkg.download_size.v;
+
+                if (reason == "Dependency") {
+                    if (data.missing_names.indexOf(name) == -1)
+                        data.extra_names.push(name);
+                }
+
+                // TODO: support removals
+            }
+        }
+
+        function signal_emitted(_path: string, _iface: string, _signal: string, _args: unknown[]) {
+            if (progress_cb) {
+                progress_cb({
+                    waiting: false,
+                    percentage: 0,
+                    cancel: null,
+                });
+            }
+        }
+
+        // TODO: decorator / helper for opening a session?
+        let session: string | null = null;
+        const client = dbus_client();
+        const subscription = client.subscribe({}, signal_emitted);
+
+        try {
+            session = await open_session();
+            await refresh(session);
+            await resolve(session);
+            await simulate(session);
+            await close_session(session);
+        } catch (err) {
+            console.warn(err);
+            if (session)
+                await close_session(session);
+            throw err;
```

Comment:
These 5 added lines are not executed by any test.  [Details](https://logs-cockpit.apps.ocp.cloud.ci.centos.org/pull-22397-64d6a5e5-20250908-152858-fedora-42-devel/Coverage/lcov/github-pr.diff.gcov.html)